res/controllers/Hercules-DJControl-Inpulse-500-script.js

```diff
@@ -0,0 +1,1566 @@
+
+  DJCi500.fxEnabledIndicator(0, "[QuickEffectRack1_" + group + "]", 0, 0);
+
+  // Slicer
+  switch(group) {
+    case "[Channel1]":
+      DJCi500.slicerBeat1.disconnect();
+      DJCi500.slicerBeat1 = engine.makeConnection('[Channel1]', 'beat_active', DJCi500.slicerBeatActive);
+      DJCi500.slicerBeat1.trigger();
+      break;
+    case "[Channel2]":
+      DJCi500.slicerBeat2.disconnect();
+      DJCi500.slicerBeat2 = engine.makeConnection('[Channel2]', 'beat_active', DJCi500.slicerBeatActive);
+      DJCi500.slicerBeat2.trigger();
+      break;
+    case "[Channel3]":
+      DJCi500.slicerBeat1.disconnect();
+      DJCi500.slicerBeat1 = engine.makeConnection('[Channel3]', 'beat_active', DJCi500.slicerBeatActive);
+      DJCi500.slicerBeat1.trigger();
+      break;
+    case "[Channel4]":
+      DJCi500.slicerBeat2.disconnect();
+      DJCi500.slicerBeat2 = engine.makeConnection('[Channel4]', 'beat_active', DJCi500.slicerBeatActive);
+      DJCi500.slicerBeat2.trigger();
+      break;
+  };
+}
+
+// This is where we choose the channel using the FX buttons and light them
+// up correctly
+DJCi500.deckSelector = function(channel, control, value, status, group) {
+  if (value === 0x7F) {
+    var deckChosen = control - 0x13;   // FX1 is 0x14, so this will yield the number
+    switch (deckChosen) {
+      case 1:
+        DJCi500.deckA.setCurrentDeck("[Channel1]");
+        DJCi500.updateDeckStatus("[Channel1]");
+        midi.sendShortMsg(0x90, 0x14, 0x7F);
+        midi.sendShortMsg(0x90, 0x16, 0x00);
+        break;
+      case 2:
+        DJCi500.deckB.setCurrentDeck("[Channel2]");
+        DJCi500.updateDeckStatus("[Channel2]");
+        midi.sendShortMsg(0x90, 0x15, 0x7F);
+        midi.sendShortMsg(0x90, 0x17, 0x00);
+        break;
+      case 3:
+        DJCi500.deckA.setCurrentDeck("[Channel3]");
+        DJCi500.updateDeckStatus("[Channel3]");
+        midi.sendShortMsg(0x90, 0x14, 0x00);
+        midi.sendShortMsg(0x90, 0x16, 0x7F);
+        break;
+      case 4:
+        DJCi500.deckB.setCurrentDeck("[Channel4]");
+        DJCi500.updateDeckStatus("[Channel4]");
+        midi.sendShortMsg(0x90, 0x15, 0x00);
+        midi.sendShortMsg(0x90, 0x17, 0x7F);
+        break;
+    };
+  };
+};
+
+DJCi500.updateEffectStatus = function(midiChannel, channel) {
+  let status = false;
+  for (var i = 1; i <= 3; i++) {
+    status = status || engine.getValue("[EffectRack1_EffectUnit" + midiChannel + "_Effect" + i + "]", "enabled");
+  }
+  return status;
+  // return engine.getValue("[EffectRack1_EffectUnit" + midiChannel + "]", "group_[Channel" + channel + "]_enable");
+}
+
+///////////////////////////////////////////////////////////////
+//                          SLICER                           //
+///////////////////////////////////////////////////////////////
+DJCi500.slicerButtonFunc = function(channel, control, value, status, group) {
+  var index = control - 0x20,
+    deck = parseInt(group.charAt(8)) - 1,

```

Comment:
Suggestion: Replace custom code with built-in [`script.deckFromGroup` helper function](https://github.com/mixxxdj/mixxx/wiki/MIDI-Scripting#helper-functions) and rename `deck` to `deckIndex` when its 0-based instead of 1-based.
```suggestion
    deck = script.deckFromGroup(deckData.currentDeck) - 1,
```